packages/slice-machine/src/hooks/usePersistedState.ts

```diff
@@ -1,27 +1,40 @@
 import { Dispatch, SetStateAction, useEffect, useState } from "react";
+import { AnyZodObject } from "zod";
 
 type PersistedState<T> = [T, Dispatch<SetStateAction<T>>];
 
 const SLICE_MACHINE_STORAGE_PREFIX = "slice-machine";
 
+type UsePersistedStateOptions = {
+  schema?: AnyZodObject;
```

Comment:
This must match T, you can use something like `ZodType<T, z.ZodTypeDef, unknown>` (maybe there's a better way, but we've been using this one in other codebases). See https://github.com/prismicio/editor/blob/df9589d286ec095ae29f17115074b5241b1721e0/packages/editor-support/src/Http.ts#L109.

Reply:
Makes sense, thanks!